File: macos/tanakaanime-macOS/VideoPlayerViewManager.m
//
//  VideoPlayerViewManager.m
//  tanakaanime-macOS
//
//  Created by Tanaka Mazivanhanga on 3/15/21.
//

#import "React/RCTViewManager.h"
@interface RCT_EXTERN_MODULE(VideoPlayerViewManager, RCTViewManager)
RCT_EXPORT_VIEW_PROPERTY(url, NSString)
RCT_EXPORT_VIEW_PROPERTY(videoName,NSString)
RCT_EXPORT_VIEW_PROPERTY(thumbnailUrl,NSString)
RCT_EXPORT_VIEW_PROPERTY(onPlayerUpdate, RCTDirectEventBlock)
RCT_EXTERN_METHOD(playFromManager:(nonnull NSNumber *)node)
RCT_EXTERN_METHOD(pauseFromManager:(nonnull NSNumber *)node)
RCT_EXTERN_METHOD(goForwardFiveFromManager:(nonnull NSNumber *)node)
RCT_EXTERN_METHOD(goBackFiveFromManager:(nonnull NSNumber *)node)
RCT_EXTERN_METHOD(seekToFromManager:(nonnull NSNumber *)node
                  seconds: (nonnull NSNumber *)seconds)



@end

File: macos/tanakaanime-macOS/VideoPlayerViewManager.swift
//
//  VideoPlayerViewManager.swift
//  tanakaanime-macOS
//
//  Created by Tanaka Mazivanhanga on 3/13/21.
//

import Foundation
import AVKit
@objc(VideoPlayerViewManager)
class VideoPlayerViewManager: RCTViewManager {
  override static func requiresMainQueueSetup() -> Bool {
    return true
  }
  override func view() -> NSView! {
    return VideoPlayerView()
  }

  @objc func playFromManager(_ node: NSNumber){
    DispatchQueue.main.async {
      guard let component = self.bridge.uiManager.view(forReactTag: node) as? VideoPlayerView else { return }
      component.play()
    }
  }

  @objc func pauseFromManager(_ node: NSNumber){
    DispatchQueue.main.async {
      guard let component = self.bridge.uiManager.view(forReactTag: node) as? VideoPlayerView else { return }
      component.pause()
    }
  }

 @objc func goBackFiveFromManager(_ node: NSNumber){
  DispatchQueue.main.async {
     guard let component = self.bridge.uiManager.view(forReactTag: node) as? VideoPlayerView else { return }
     component.goBackFive()
   }
  }
 @objc func goForwardFiveFromManager(_ node: NSNumber){
  DispatchQueue.main.async {
     guard let component = self.bridge.uiManager.view(forReactTag: node) as? VideoPlayerView else { return }
     component.goForwardFive()
   }
  }
  @objc func seekToFromManager( _ node: NSNumber, seconds: NSNumber){
    DispatchQueue.main.async {
       guard let component = self.bridge.uiManager.view(forReactTag: node) as? VideoPlayerView else { return }
       component.seekTo(seconds)
     }
  }
}

